classes/journal/JournalDAO.inc.php

```diff
@@ -418,6 +422,67 @@ function resequenceJournals() {
 	function getInsertJournalId() {
 		return $this->getInsertId('journals', 'journal_id');
 	}
+	
+	/**
+	 * Get journals by setting.  Backported from master pkp-lib's ContextDAO.
+	 * @param $settingName string
+	 * @param $settingValue mixed
+	 * @param $contextId int
+	 * @return DAOResultFactory
+	 */
+	function getBySetting($settingName, $settingValue, $contextId = null) {
+		$params = array($settingName, $settingValue);
+		if ($contextId) $params[] = $contextId;
+
+		$result = $this->retrieve(
+			'SELECT * FROM ' . $this->_getTableName() . ' AS c
+			LEFT JOIN ' . $this->_getSettingsTableName() . ' AS cs
+			ON c.' . $this->_getPrimaryKeyColumn() . ' = cs.' . $this->_getPrimaryKeyColumn() .
+			' WHERE cs.setting_name = ? AND cs.setting_value = ?' .
+			($contextId?' AND c.' . $this->_getPrimaryKeyColumn() . ' = ?':''),
+			$params
+		);
+
+		return new DAOResultFactory($result, $this, '_fromRow');
+	}
+
+	/**
+	 * Internal function to return a Journal object from a row.  Backported from master OJS's JournalDAO.
+	 * @param $row array
+	 * @return Journal
+	 */
+	function _fromRow($row) {
```

Comment:
I wanted to keep getBySetting() aligned to ContextDAO::getBySetting() for the sake of porting any future changes introduced in master's ContextDAO.  Thus the raw copy, including https://github.com/pkp/ojs/pull/502/files#r30907959 and https://github.com/pkp/ojs/pull/502/files#r30908188.  Still want to make this diverge?